ngraph/core/src/op/util/nms_base.cpp

```diff
@@ -0,0 +1,188 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ngraph/op/util/nms_base.hpp"
+#include <cstring>
+#include <ngraph/validation_util.hpp>
+#include "itt.hpp"
+#include "ngraph/attribute_visitor.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/util/op_types.hpp"
+#include "ngraph/type/bfloat16.hpp"
+#include "ngraph/type/float16.hpp"
+#include "ngraph/util.hpp"
+
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(op::util::NmsBase, "NmsBase", 0);
+
+op::util::NmsBase::NmsBase(const Output<Node>& boxes,
+                           const Output<Node>& scores,
+                           const SortResultType sort_result_type,
+                           const ngraph::element::Type& output_type,
+                           const int nms_top_k,
+                           const int keep_top_k)
+    : Op({boxes, scores})
+    , m_sort_result_type{sort_result_type}
+    , m_output_type{output_type}
+    , m_nms_top_k{nms_top_k}
+    , m_keep_top_k{keep_top_k}
+{
+}
+
+namespace
+{
+    inline bool is_float_type_admissible(const element::Type& t)
+    {
+        return t == element::f32 || t == element::f16 || t == element::bf16;
+    }
+} // namespace
+
+void op::util::NmsBase::validate()
+{
+    const auto boxes_ps = get_input_partial_shape(0);
```

Comment:
Please add `NGRAPH_OP_SCOPE` to this method

Reply:
Done